Document declarative configuration of Test Visibility in Jenkins using 'datadog' step (#21241)
DataDog/documentation: content/en/continuous_integration/pipelines/jenkins.md

See the [Test Visibility documentation][17] for your language to make sure that the testing framework that you use is supported.

There are different ways to enable Test Visibility inside a Jenkins job or pipeline:

> 1. Using Jenkins configuration UI.

Reviewer: ```suggestion
1. Using the Jenkins configuration UI.
```

> 2. Adding `datadog` step inside the pipeline script.

Reviewer: ```suggestion
2. Adding the `datadog` step inside the pipeline script.
```

3. Configuring the tracer manually.

### Enable with the Jenkins configuration UI

UI-based Test Visibility configuration is available in Datadog Jenkins plugin v5.6.0 or newer.

> This option is not suitable for pipelines that are configured entirely in `Jenkinsfile` (for example, Multi-Branch pipelines or pipelines from Organization Folder).

Reviewer: ```suggestion
This option is not suitable for pipelines that are configured entirely in `Jenkinsfile` (for example, Multibranch pipelines or pipelines from Organization Folder).
```

> For these pipelines use declarative configuration with `datadog` step (described in the next section).

Reviewer: ```suggestion
For these pipelines use declarative configuration with the `datadog` step (described in the next section).
```

{{< img src="ci/ci-jenkins-plugin-tests-config.png" alt="Datadog Test Visibility configuration for Jenkins" style="width:100%;">}}

> ### Enable with datadog pipeline step

Reviewer: ```suggestion
### Enable with the `datadog` pipeline step
```

This configuration option is available in Datadog Jenkins plugin v5.6.2 or newer.

In declarative pipelines, add the step to a top-level `options` block like so:

```groovy
pipeline {
    agent any
    options {
        datadog(testVisibility: [
            enabled: true,
            serviceName: "my-service", // the name of service or library being tested
            languages: ["JAVA"], // languages that should be instrumented (available options are "JAVA", "JAVASCRIPT", "PYTHON")
            additionalVariables: ["my-var": "value"]  // additional tracer configuration settings (optional)
        ])
    }
    stages {
        stage('Example') {
            steps {
                echo "Hello world."
            }
        }
    }
}
```

> In scripted pipeline, wrap the relevant section with the datadog step like so:

Reviewer: ```suggestion
In a scripted pipeline, wrap the relevant section with the `datadog` step like so:
```

```groovy
datadog(testVisibility: [ enabled: true, serviceName: "my-service", languages: ["JAVASCRIPT"], additionalVariables: [:] ]) {
  node {
    stage('Example') {
      echo "Hello world."
    }
  }
}
```

> The other datadog settings, such as `collectLogs` or `tags` can be added alongside the `testVisibility` block.

Reviewer: ```suggestion
The other `datadog` settings, such as `collectLogs` or `tags` can be added alongside the `testVisibility` block.
```
